homeassistant/components/climate/spider.py

```diff
@@ -0,0 +1,151 @@
+"""
+Support for Spider thermostats.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/climate.spider/
+"""
+
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.climate import (
+    ATTR_TEMPERATURE, PLATFORM_SCHEMA, STATE_COOL, STATE_HEAT, STATE_IDLE,
+    SUPPORT_TARGET_TEMPERATURE, ClimateDevice)
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME, TEMP_CELSIUS
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['itho_daalderop_api==1.0.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=15)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Spider thermostat."""
+    from itho_daalderop_api import IthoDaalderop_API, UnauthorizedException
+
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+
+    try:
+        api = IthoDaalderop_API(username, password)
+
+    except UnauthorizedException:
+        _LOGGER.exception("Can't connect to the Itho Daalderop API")
+        return
+
+    thermostats = api.get_thermostats()
+    for i, thermostat in enumerate(thermostats):
+        add_devices(
+            [SpiderThermostat(api, thermostat['id'],
+                              thermostat['name'], i == 0)],
+            True
+        )
+
+
+class SpiderThermostat(ClimateDevice):
+    """Representation of a thermostat."""
+
+    def __init__(self, client, thermostat_id, name, master):
+        """Initialize the thermostat."""
+        self.client = client
+        self._id = thermostat_id
+        self._name = name
+        self._master = master
+        self._thermostat = None
+        self._current_temperature = None
+        self._target_temperature = None
+        self._min_temp = None
+        self._max_temp = None
+        self._operation = STATE_IDLE
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        supported = (SUPPORT_TARGET_TEMPERATURE)
```

Comment:
Just return the constant here.

Reply:
Is it not possible to change the operation mode?

Reply:
operation mode can only be changed on the master. Added this now.